Completed API - Alpha Release

robertmclaws opened the pull request:
With this PR, the API _theoretically_ functions end-to-end. It is a fully-queued, fully-reactive framework. 

Next steps:
- Build unit tests against a self-hosted WebAPI endpoint that can just accept requests and return fake responses.
- Wire it through the GameServices to the app, demonstrating login and firing some other commands.

Still needed:
- Duplicate request safeguards.

Commit:
Major work on new pipeline

- Eliminated ApiFailureStrategy.
- New queued pipeline almost finished.
- Basic events work started.

Commit:
Continuing work from last night.

- Moving more Interfaces as needed.
- Deleting ApiFailureStrategy and RetryHandler.
- Making PostProto recursive.

Commit:
Clients Refactor, Part 1

Too much to mention.

Commit:
EncounterClient Refactor

Commit:
CancellationToken enhancements

- Weave the CancellationToken further into the request, so the
currently-executing request can also be cancelled.

Commit:
Wire Queue & Response Processing

- Wire CancellationToken directly to HttpClient request.
- Added logging for reasons request processing may be halted.
- Wire all this code into `StartProcessingRequests` and wire the output
to the response processor.

Commit:
Refactor InventoryClient

Commit:
Hercules: The Refactor Continues

Commit:
Too many changes to mention

Changed: Everything.

Commit:
Mapping RequestType to ResponseMessages.

Commit:
T4 All The Things!

- T4 code generation template to generate the necessary events that the
Client will throw.
- Never write 700 lines of code when you can write 65 lines of code that
writes 700 lines of code.

Commit:
Slight cleanup.

Commit:
Add Event firing to the T4 template.

- With this checkin, the API should function. Time to move to unit
tests.

Commit:
Ensure requests are properly formatted.

review-ninja commented:
<a href="https://app.review.ninja/PoGo-Devs/PoGo/pull/28" target="_blank"><img src="https://app.review.ninja/65663167/pull/28/badge" alt="ReviewNinja"/>

Commit:
Halt requests on `AuthenticateAsync`

Authenticate should only be called on a clean login, so if we're
re-authenticating for some reason, make sure all requests are killed
ASAP.

Commit:
Called the. cancel function instead

... in case we decide to throw a Cancelling event or something.

WallyCZ commented on src/PoGo.ApiClient/PokemonGoApiClient.Events.Generated.cs:
```diff
@@ -0,0 +1,888 @@
+            {
+				switch (message.GetType().Name)
+				{
+					case "AddFortModifierResponse":
```
Could be used nameof(AddFortModifierResponse)?

robertmclaws replied:
Great fix! Will put it in the next PR.
